src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/gadgets/MultiTool.java

```diff
@@ -73,25 +78,49 @@ protected ItemUseHandler getItemUseHandler() {
         return e -> {
             Player p = e.getPlayer();
             ItemStack item = e.getItem();
+            PersistentDataContainer container = item.getItemMeta().getPersistentDataContainer();
             e.cancel();
-
-            int index = selectedMode.getOrDefault(p.getUniqueId(), 0);
-
-            if (!p.isSneaking()) {
-                if (removeItemCharge(item, COST)) {
-                    SlimefunItem sfItem = modes.get(index).getItem();
-
-                    if (sfItem != null) {
-                        sfItem.callItemHandler(ItemUseHandler.class, handler -> handler.onRightClick(e));
+            
+            int index = 0;
+            if (container.has(key, PersistentDataType.STRING)) {
+                for (int i = 0; i < modes.size(); i++) {
+                    if (modes.get(i).getItemId() == container.get(key, PersistentDataType.STRING)) {
+                        index = i;
+                        break;
                     }
                 }
+            }
+
+            SlimefunItem sfItem = modes.get(index).getItem();
+
+            if (!p.isSneaking() && removeItemCharge(item, COST) && sfItem != null) {
+                sfItem.callItemHandler(ItemUseHandler.class, handler -> handler.onRightClick(e));
             } else {
                 index = nextIndex(index);
 
                 SlimefunItem selectedItem = modes.get(index).getItem();
                 String itemName = selectedItem != null ? selectedItem.getItemName() : "Unknown";
                 Slimefun.getLocalization().sendMessage(p, "messages.multi-tool.mode-change", true, msg -> msg.replace("%device%", "Multi Tool").replace("%mode%", ChatColor.stripColor(itemName)));
-                selectedMode.put(p.getUniqueId(), index);
+
+                ItemMeta meta = item.getItemMeta();
+                List<String> lore = meta.hasLore() ? meta.getLore() : new ArrayList<>();
+
+                meta.getPersistentDataContainer().set(key, PersistentDataType.STRING, modes.get(index).getItemId());
```

Comment:
Again you can use PersistentDataAPI

Reply:
I thought using a Persistent Data Container here would be enough, also I saw PDCs being used this way in many places in Slimefun. Can you give me more guidance on how it should be implemented?

Reply:
PersistentDataAPI is a helper, you can use it like `PersistentDataAPI.setString(meta, key, modes.get(index).getId())`. It's just a nicer API

Reply:
Oh okay, I'll use it then. Thanks.